lib/ecto/adapters/sql.ex

```diff
@@ -141,11 +146,17 @@ defmodule Ecto.Adapters.SQL do
   def to_sql(kind, repo, queryable) do
     adapter = repo.__adapter__
 
-    {_meta, prepared, params} =
+    plan =
       Ecto.Queryable.to_query(queryable)
       |> Ecto.Query.Planner.query(kind, repo, adapter)
-
-    {prepared, params}
+    case plan do
+      {:execute, _meta, %{statement: statement}, params} ->
```

Comment:
One issue we have that is confusing is that are so many different ways to prepare now